### ISSUE TITLE
gRPC streaming issue with linkerd 1.1.3

### ISSUE THREAD

**Issue by vadimi** [linkerd/linkerd#1591]
Hello,

I'm experiencing an issue with gRPC streaming in linkerd `1.1.3`. I have a gRPC server that sends a few thousand messages to a stream and a client that consumes them. After a few hundred messages (depends on message size) linkerd stops accepting new ones and my server and client just hang (or timeout expires). It does not happen on small streams (a couple of messages), also linkerd `1.1.2` works fine, I can send as many messages as I want.

**Comment by klingerf** [comment]
Hey @vadimi, thanks for reporting this. Am seeing something similar with 1.1.3 in one of our test environments:

![screen shot 2017-08-16 at 12 06 26 pm](https://user-images.githubusercontent.com/9226/29424379-54f49bde-8334-11e7-8d0d-ed366a367417.png)

We should be able to reproduce and figure out what's going. Can you tell us a bit more about the environment where you're running linkerd? Is it Kubernetes? Are you using namerd? If you don't mind pasting your full linkerd config into this issue that would be super helpful. Thanks!

**Comment by vadimi** [comment]
We run linkerd in DC/OS in production. But I was able to reproduce it in my local dev environment by just running linkerd docker container. Here is the config I used:

```
admin:
  port: 9990

routers:
- protocol: h2
  experimental: true
  identifier:
    kind: io.l5d.header.token
  dtab: |
    /ph        => /$/io.buoyant.rinet ;
    /svc       => /ph/80 ;
    /srv       => /$/io.buoyant.porthostPfx/ph ;
    /svc       => /srv ;
  servers:
  - port: 4140
    ip: 0.0.0.0
```

and docker run command:

```
docker run --rm --name linkerd -p 9990:9990 -p 4140:4140 --network=host -v `pwd`/config.yaml:/config.yaml buoyantio/linkerd:1.1.3 -log.level=DEBUG /config.yaml
```

**Comment by klingerf** [comment]
Great, thanks for the additional info. I was able to reproduce as well. Here are metrics and thread dumps from a linkerd process after it gets into this state:

* [grpc-metrics.txt](https://github.com/linkerd/linkerd/files/1232635/grpc-metrics.txt)
* [grpc-thread-dump.txt](https://github.com/linkerd/linkerd/files/1232636/grpc-thread-dump.txt)

I also checked the linkerd debug logs and didn't see anything obvious that would cause this.

**Comment by hawkw** [comment]
@adleong, this was fixed in #1598, correct?

**Comment by adleong** [comment]
Correct!
